### ISSUE TITLE
Last-position-mark is set a step late

### ISSUE THREAD

**Issue by BohrShaw** [akhodakivskiy/VimFx#717]
Steps to reproduce:

1. Scroll a page down
2. Press `gg`
3. Press ``` `` ```

Expected result: Return to the position *before* the last jump `gg`, IE. the position after step 1.

Actual result: It's at the top of the page.

I think the expected behavior is more useful and it is similar in Vim.

VimFx version: 0.13.1
Firefox version: 45.0.1
OS: Manjaro(Xfce)
Keyboard language: EN_US

**Comment by lydell** [comment]
Sorry, I can’t reproduce this.

**Comment by BohrShaw** [comment]
On this page, press ```Ggg`` ```, and I'm on the top of the page, which I expect to be on the bottom.
Do you get at the bottom?

**Comment by lydell** [comment]
Yes.

**Comment by BohrShaw** [comment]
I tested in a new profile and it worked as expected. But after I sourced my [config file](https://github.com/BohrShaw/vish/blob/master/vimfx/config.js), this issue occurred. Can you help me inspect what's possibly wrong with my config.

**Comment by lydell** [comment]
As far as I can tell by looking at your config file, `` Ggg`' `` and `Ggg''` should take you to the last scroll position, while ``` Ggg`` ``` should display a "No mark for: `" notification (unless you've run `` m` `` before).

**Comment by BohrShaw** [comment]
This issue applies when I press `` `' `` or `''`. And I have reset the mark related options but it still doesn't work. You may apply my config file and see what's wrong.

**Comment by lydell** [comment]
Comment out code in your config file until you find the minimal lines that produces your problem.

**Comment by BohrShaw** [comment]
How can quickly I reset all options?

**Comment by BohrShaw** [comment]
How can I quickly reset all options?

**Comment by lydell** [comment]
1. Go to `about:config`.
2. Filter on `extensions.VimFx.`.
3. Sort on the "Status" column (so that modified prefs com first).
4. Select the first item.
5. Press the `<menu>` key to open the context menu.
6. Press the access key for the "Reset" item ("r" in English).
7. Press `<home>` to select the first item.
8. Repeat steps 5 to 7 until satisfied.

**Comment by BohrShaw** [comment]
I've reset all VimFx options and this issue still bites me, even after I fully reload a page.

Then it might be related to Firefox settings in `about:preferences`, and is not related to my addons as I've tested it in a new clean profile.

**Comment by lydell** [comment]
Run the following in the browser console to reset all VimFx prefs:

```js
branch=Services.prefs.getBranch('extensions.VimFx.'); branch.getChildList('').forEach(pref => branch.clearUserPref(pref))
```

**Comment by BohrShaw** [comment]
In a new clean profile, I installed VimFx, sourced my config file, removed VimFx and reinstalled. And I still encounter this issue.

**Comment by lydell** [comment]
So you mean that as soon as you've sourced your config file once, the Firefox profile in question gets "broken"?

**Comment by BohrShaw** [comment]
Yes

**Comment by lydell** [comment]
I just did the following:

1. Created a new profile in Firefox 45.
2. Installed VimFx 0.13.1.
3. Tried the ``` Ggg`` ``` dance. It worked as expected.
4. Source your config file.
5. Tried the ``` Ggg`` ``` dance again. It worked as expected this time too.

I did the above with both multi-process enabled and disabled.

I really have no idea what's going on for you :(

Try sourcing a completely empty config file in a new profile and see if the problem occurs then too.
